platform/android/src/jni.cpp

```diff
@@ -373,6 +435,77 @@ std::pair<mbgl::AnnotationSegment, mbgl::ShapeAnnotation::Properties> annotation
     return std::make_pair(segment, shapeProperties);
 }
 
+/*
+ * Adapted from https://github.com/dropbox/djinni/blob/master/support-lib/jni/Marshal.hpp
+ */
+
+void jniExceptionCheck(JNIEnv * env) {
+    if (!env) {
+        mbgl::Log::Error(mbgl::Event::JNI, "Empty environment.");
+        abort();
+    }
+
+    const jthrowable e(env->ExceptionOccurred());
+    if (e) {
+        env->ExceptionClear();
```

Comment:
I've removed this code that was coming from Dropbox's djinni and redone it in https://github.com/mapbox/mapbox-gl-native/commit/7f7f492219c29e800d190d54aa656c984a917538.